disable building channelz if gRPC_USE_PROTO_LITE

yorkie opened the pull request:
See https://github.com/grpc/grpc/issues/19473

hcaseyal commented:
@jtattermusch I'd like your input on https://github.com/grpc/grpc/issues/19472 before considering the changes suggested in this PR

yorkie commented:
@hcaseyal Thanks for your reviews, and I have achieved the followings:

- signed the GRPC  CLA
- removed the previous changes on auto-generated contents
- introduced a new attribute `extra_cond` for libs, and updated the CMakeFile template
  - updated the `build.yaml` for building channelz only if not `gRPC_USE_PROTO_LITE`

yorkie commented:
I still have a question, why do we maintain the generated file on the tree? Because I'm confused that if should generate once again those files and push them on this PR?

jtattermusch commented on build.yaml:
```diff
@@ -2110,6 +2110,8 @@ libs:
   - grpc
   filegroups:
   - grpcpp_channelz_proto
+  extra_cond:
```
This seems pretty adhoc, and also in the long run we'll be getting rid of build.yaml, so it doesn't make sense to add new properties like this.
We could just specialcase directly in CMakeLists.txt.template if needed, but I'm not sure it's worth doing it considering that gRPC_USE_PROTO_LITE is probably used only by few.

jtattermusch commented:
It looks like due to https://github.com/grpc/grpc/issues/19472, `protoc` won't work when gRPC_USE_PROTO_LITE is enabled anyway (generated .proto code is needed for channelz_service to build), so this PR cannot really fix the problem?

jtattermusch commented:
I posted a workaround in https://github.com/grpc/grpc/issues/19473#issuecomment-552382027 (and I think this PR brings too much complexity for something that not really used by many).

jtattermusch commented:
Closing in favor of https://github.com/grpc/grpc/pull/21011
